Description: build(meson): use C++14 with GTest >= 1.13.0
 GoogleTest, starting with version 1.13.0, requires C++14 to build. This
 patch enables C++14 if GoogleTest 1.13.0 or newer is detected when
 compiling the tests with Meson, making it possible to use new GTest
 versions.
Author: Andrea Pappacoda <andrea@pappacoda.it>
Origin: upstream, https://github.com/yhirose/cpp-httplib/commit/52d8dd41f1dc239d8eceed1253631022653783cc
Reviewed-By: <name and email of someone who approved/reviewed the patch>
Last-Update: 2023-07-13

--- cpp-httplib-0.11.4+ds.orig/test/meson.build
+++ cpp-httplib-0.11.4+ds/test/meson.build
@@ -86,6 +86,12 @@ subdir(join_paths('www', 'dir'))
 subdir(join_paths('www2', 'dir'))
 subdir(join_paths('www3', 'dir'))
 
+# GoogleTest 1.13.0 requires C++14
+test_options = []
+if gtest_dep.version().version_compare('>=1.13.0')
+  test_options += 'cpp_std=c++14'
+endif
+
 test(
   'main',
   executable(
@@ -94,7 +100,8 @@ test(
     dependencies: [
       cpp_httplib_dep,
       gtest_dep
-    ]
+    ],
+    override_options: test_options
   ),
   depends: [
     key_pem,
